### ISSUE TITLE
Cargo gets private gitlab repository code failed by git-credentials store

### ISSUE THREAD

**Issue by tensorchen** [rust-lang/rust#66803]
**Problem**

Cargo gets private gitlab repository code failed，git-credentials not work，write username/password in cargo.toml is ok.

_git-credentials not work_
```bash
$ cat ~/.gitconfig
[credential]
        helper = store

$ cat ~/.git-credentials
http://tensorchen:xxx_mypassword@git.code.oa.com

$ cat Cargo.toml
[package]
name = "xxxxx"
version = "0.1.0"
edition = "2018"

[dependencies]
l5 = {git = "http://git.code.oa.com/birdio/l5-rs", branch = "master"}

$ cargo build 
    Updating git repository `http://git.code.oa.com/birdio/l5-rs`
warning: spurious network error (2 tries remaining): failed to receive HTTP 200 response: got 401; class=Net (12)
warning: spurious network error (1 tries remaining): failed to receive HTTP 200 response: got 401; class=Net (12)
error: failed to load source for a dependency on `l5`

Caused by:
  Unable to update http://git.code.oa.com/birdio/l5-rs

Caused by:
  failed to fetch into /root/.cargo/git/db/l5-rs-56b3f1291d851527

Caused by:
  failed to receive HTTP 200 response: got 401; class=Net (12)
```

_write username/password in cargo.toml is ok_

```bash
$ cat Cargo.toml
[dependencies]
l5 = {git = "http://tensorchen:xxx_mypassword@git.code.oa.com/birdio/l5-rs", branch = "master"}

$ cargo build 
Updating git repository `http://tensorchen:xxxx@git.code.oa.com/birdio/l5-rs`
   Compiling l5 v0.1.0 (http://tensorchen:xxxxx@git.code.oa.com/birdio/l5-rs#39fffd34)
   Compiling rust-by-example v0.1.0 (/root/git.code.oa.com/tensorchen/rust-by-example)
    Finished dev [unoptimized + debuginfo] target(s) in 0.52s
```

**Notes**

```bash
$ cargo version
cargo 1.39.0 (1c6ec66d5 2019-09-30)
$ git version
git version 2.23.0
```

**Comment by sfackler** [comment]
This should be filed on the rust-lang/cargo repository.

**Comment by bjorn3** [comment]
Maybe libgit2 doesn't support git-credentials?

**Comment by csmoe** [comment]
Change the git url to SSH (`ssh://...`) may help here. 
And this issue belongs to https://github.com/rust-lang/cargo , please file this there.
